src/main/java/io/spokestack/spokestack/nlu/tensorflow/TFNLUOutput.java

```diff
@@ -64,21 +64,27 @@ public void registerSlotParsers(Map<String, SlotParser> parsers) {
         String[] tagLabels = getLabels(output, numTokens);
         context.traceDebug("Tag labels: %s", Arrays.toString(tagLabels));
         Map<Integer, Integer> slotLocations = new HashMap<>();
-        Integer curSlotStart = null;
+        Tuple<String, Integer> curSlot = null;
         for (int i = 0; i < tagLabels.length; i++) {
             String label = tagLabels[i];
             if (label.equals("o")) {
-                if (curSlotStart != null) {
-                    curSlotStart = null;
+                if (curSlot != null) {
+                    slotLocations.put(curSlot.second(), i);
```

Comment:
why are you tracking `o` tags now?

Reply:
That's not tracking `o`s; it's saying "if you're in the middle of a tag sequence when you hit an `o`, record the `o` as the end of that sequence and note that you're no longer in one".

Reply:
(remember the half-open interval nature of sublists in Java; recording the `i` index as the end of the interval means "the sequence stops right before this index")

Reply:
thanks for explaining

Reply:
NP; thanks for keeping me honest.